src/sdk/util.hpp

```diff
@@ -0,0 +1,155 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef realm_sdk_util_h
+#define realm_sdk_util_h
+
+#include <cxxabi.h>
+#include <type_traits>
+#include "property.hpp"
+
+namespace realm::sdk::util {
+
+namespace type_traits {
+struct embedded_object;
+struct object;
+struct list;
+struct primitive;
+struct enum_type;
+struct primary_key;
+} // namespace type_traits
+
+template <typename T>
+using EmbeddedObject = Property<T, util::type_traits::embedded_object>;
+
+static inline std::string demangle(const char* name) {
+    int status = -4; // some arbitrary value to eliminate the compiler warning
+
+    std::unique_ptr<char, void(*)(void*)> res {
+        abi::__cxa_demangle(name, NULL, NULL, &status),
```

Comment:
Does this sort of call guarantee consistency between different platforms and vendors?